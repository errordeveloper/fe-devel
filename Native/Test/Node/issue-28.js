/*
 *  Copyright 2010-2012 Fabric Engine Inc. All rights reserved.
 */

F = require('Fabric').createClient();

op = F.DG.createOperator("videoSeekTime");
op.setSourceCode(
'require FabricVIDEO;\n'+
'operator videoSeekTime(\n'+
'  io VideoHandle handle,\n'+
'  io Scalar time,\n'+
'  io RGB pixels[],\n'+
'  io RGB cache[][Integer]\n'+
'  ) {\n'+
'  Integer frameIndex = floor(time * handle.fps + 0.5);\n'+
'  if(cache.has(frameIndex)) {\n'+
'    pixels = cache[frameIndex];\n'+
'    return;\n'+
'  }\n'+
'\n'+
'  FabricVIDEOSeekTime(handle, time);\n'+
'  FabricVIDEOGetAllPixels(handle, pixels);\n'+
'\n'+
'  cache[frameIndex] = pixels;\n'+
'}\n'
);
op.setEntryPoint("videoSeekTime");
if (op.getErrors().length > 0) {
  console.log(op.getErrors());
  if (op.getDiagnostics().length > 0) {
    console.log(op.getDiagnostics());
  }
}

F.close();

