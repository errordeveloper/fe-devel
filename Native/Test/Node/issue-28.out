[ 'compile failed' ]
[ { filename: '(unknown)',
    line: 9,
    column: 6,
    level: 'error',
    desc: 'no such method RGB[][Integer].has(io Integer)' },
  { filename: '(unknown)',
    line: 17,
    column: 3,
    level: 'error',
    desc: 'no cast exists from RGB[] to RGB[Integer]' } ]
